swift/ITSClient/Sources/ITSMobility/Models/ETSI.swift

```diff
@@ -0,0 +1,105 @@
+    private static let decimetersPerSquaredSecondFactor: Double = 10
+
+    static func degreesToDeciMicroDegrees(_ degrees: Double) -> Int {
+        Int(degrees * deciMicroDegreesFactor)
+    }
+
+    static func deciMicroDegreesToDegrees(_ deciMicroDegrees: Int) -> Double {
+        Double(deciMicroDegrees) / deciMicroDegreesFactor
+    }
+
+    static func epochTimestampToETSIMilliseconds(_ epochTimestamp: TimeInterval) -> Int {
+        Int(Int(epochTimestamp * 1000) - (etsiUnixEpochDifference * 1000))
+    }
+
+    static func etsiMillisecondsToEpochTimestamp(_ etsiMilliseconds: Int) -> TimeInterval {
+        TimeInterval(etsiMilliseconds + etsiUnixEpochDifference * 1000) / 1000
+    }
+
+    static func epochTimestampToETSISeconds(_ epochTimestamp: TimeInterval) -> Int {
+        Int(epochTimestamp) - etsiUnixEpochDifference
+    }
+
+    static func secondsToCentiSeconds(_ seconds: TimeInterval) -> Int {
+        Int(seconds * centiSecondsFactor)
+    }
+
+    static func centiSecondsToSeconds(_ centiSeconds: Int) -> TimeInterval {
+        Double(centiSeconds) / centiSecondsFactor
+    }
+
+    static func metersToCentimeters(_ meters: Double) -> Int {
+        Int(meters * centimetersFactor)
+    }
+
+    static func centimetersToMeters(_ centimeters: Int) -> Double {
+        Double(centimeters) / centimetersFactor
+    }
+
+    static func metersToDecimeters(_ meters: Double) -> Int {
+        Int(meters * decimetersFactor)
+    }
+
+    static func decimetersToMeters(_ decimeters: Int) -> Double {
+        Double(decimeters) / decimetersFactor
+    }
+
+    static func metersPerSecondToCentimetersPerSecond(_ meters: Double) -> Int {
+        Int(meters * centimetersPerSecondFactor)
+    }
+
+    static func centimetersPerSecondToMetersPerSecond(_ centimetersPerSecond: Int) -> Double {
+        Double(centimetersPerSecond) / centimetersPerSecondFactor
+    }
+
+    static func degreesToDeciDegrees(_ degrees: Double) -> Int {
+        Int(degrees * deciDegreesFactor)
+    }
+
+    static func deciDegreesToDegrees(_ deciDegrees: Int) -> Double {
+        Double(deciDegrees) / deciDegreesFactor
+    }
+
+    static func degreesPerSecondToCentiDegreesPerSecond(_ degreesPerSecond: Double) -> Int {
+        Int(degreesPerSecond * centiDegreesPerSecondFactor)
+    }
+
+    static func centiDegreesPerSecondToDegreesPerSecond(_ centiDegreesPerSecond: Int) -> Double {
+        Double(centiDegreesPerSecond) / centiDegreesPerSecondFactor
+    }
+
+    static func metersPerSquaredSecondToDecimetersPerSquaredSecond(_ metersPerSquaredSecond: Double) -> Int {
+        Int(metersPerSquaredSecond * decimetersPerSquaredSecondFactor)
+    }
+
+    static func decimetersPerSquaredSecondToMetersPerSquaredSecond(_ decimetersPerSquaredSecond: Int) -> Double {
+        Double(decimetersPerSquaredSecond) / decimetersPerSquaredSecondFactor
+    }
+}
+
+@inlinable func clip<T>(_ value: T, _ minimum: T, _ maximum: T) -> T where T : Comparable {
```

Comment:
There are keys which value will be set to a magic value when it is out-of-range, not just clipped. For example, in a CAM, `message.basic_container.confidence.position_confidence_ellipse.semi_major_confidence` has a range of [0, 4093], and when the value is out-of-range, it should be set to 4094.

Reply:
Ok I set the value to unavailable in this case, I need to test the min/max value with a fallback value instead of clipping. But when a key is optional, why do we need an unavailable value ?

Reply:
It is not about an "unavailable" value; it is about an "out-of-range" value. I.e. the value is known, but can not be represented by the expected range.

Reply:
Oh, and I suggested in the review comment: since you don't need that behaviour for now, just leave it aside; we can extend `clip()` when we actually need an out-of-range indication.

Reply:
Yes I understood that's for out of a range value but I'm wondering in which case you set the unavailable value when the field is optional. For now, I don't add the key in the json.
We can add an issue to solve the out of range values later.

Reply:
For optional keys, it is OK to not emit those when they are not available, indeed.

But there are keys that are not optional, and that can be unavailable. For example, in a CAM, `message.basic_container.reference_position.latitude` is required, but can be set to `900000001` when it is unknown.

Reply:
Issue created here: https://github.com/Orange-OpenSource/its-client/issues/355